mffpy/raw_bin_files.py

```diff
@@ -34,45 +34,70 @@ def __init__(self, filepointer: IO[bytes]):
         assert not self.filepointer.closed
         self.buffering: bool = False
 
-    def __del__(self):
-        self.close()
+    def read_raw_samples(self, t0: float = 0.0,
+                         dt: float = None, block_slice: slice = None
+                         ) -> Tuple[np.ndarray, float]:
+        """return `(channels, samples)`-array and `start_time` of data
 
-    def close(self):
-        self.filepointer.close()
+        The signal data is organized in variable-sized blocks that enclose
+        epochs of continuous recordings.  Discontinuous breaks can happen in
+        between blocks.  `block_slice` indexes into such epochs if not None,
+        but we might want only a small chunk of it given by `t0` and `dt`.
+        Therefore, we further index into blocks `bsi` selected through
+        block_slice with the variables `A` and `B`.  Block indices `A` and `B`
+        are chosen to enclose the interval `(t0, t0+dt)` which we would like to
+        read.
 
-    def tell(self) -> int:
-        return self.filepointer.tell()
+        **Parameters**
+        t0: float (default: 0.0)
+            Start time to read out data, starting at the beginning of the
+            block.
+        dt: float (default: None)
+            duration of the data to read out.  `None` defaults to the rest of
+            the signal.
+        block_slice: slice (default: None)
+            blocks to consider when reading data.
 
-    def seek(self, loc, mode=SEEK_SET):
-        assert mode != SEEK_SET or loc >= 0
-        assert mode != SEEK_END or loc <= 0
-        return self.filepointer.seek(loc, mode)
+        **Returns**
+        block_data: np.ndarray
+            array containing all samples between the samples enclosing
+            `(t0,t0+dt)` relative to the block slice.
+        time_of_first_sample: float
+            time in seconds from file start of the first returned sample.
+        """
+        assert block_slice is None or isinstance(block_slice, slice)
+        block_slice = block_slice if block_slice else slice(
+            0, len(self.block_start_idx)-1)
+        # Calculate .. the relative sample index of `t0` and `t0+dt`
+        # with respect to the beginning of the epoch (block_slice)
+        sr = self.signal_blocks['sampling_rate']
+        a = np.round(t0*sr).astype(int) if t0 is not None else None
+        b = np.round((t0+dt)*sr).astype(int) if dt is not None else None
+        time_of_first_sample = a/sr
```

Comment:
If `a` is `None`, I think we will get an error here. Could we just have
```
time_of_first_sample = t0
```

Reply:
You are right, but when you think about it, you'll see that `time_of_first_sample` isn't exactly `t0` because of the rounding operation.

Reply:
Ah true. Good point. So maybe
```
time_of_first_sample = a/sr if a is not None else 0.0
```